website/content/docs/connect/ca/vault.mdx

```diff
@@ -210,48 +210,79 @@ The following Vault policy allows Consul to use pre-existing PKI paths in Vault.
 Consul is granted read-only access to the PKI mount points and the Root CA, but is
 granted full control of the Intermediate or Leaf CA for Connect clients.
 
-In this example the `RootPKIPath` is `connect_root` and the `IntermediatePKIPath`
-is `connect_inter`. These values should be updated for your environment.
+This example uses placeholder values for `RootPKIPath` and `IntermediatePKIPath`.
+Please update those placeholders to match the path values in the CA provider configuration.
 
 <CodeBlockConfig filename="vault-managed-pki-policy.hcl">
 
 ```hcl
-# Existing PKI Mounts
-path "/sys/mounts" {
-  capabilities = [ "read" ]
-}
+# -----------------------------------------------------------------------------
```

Comment:
Curious about these in-file explanations. Do they mirror a template file included in some package?

Reply:
I didn't check what established practices we have for including comments in HCL file examples. JSON doesn't support comments, so it's not even possible in that context.

I included these comments so that, if someone copies and pastes this policy for their own usage, someone looking at it now or later can understand the purpose when it might not otherwise be obvious. In my experience, I've seen configuration file templates included with software include inline docs comments like this, including commented out things with an explanation of when it should be uncommented (like I've done here for the root rotation permission).

I'm open to removing the comments. I personally found the policy a lot more understandable with them than without them. It also helps with playing "spot the difference" between the Consul vs Vault managed mounts approaches.

Before proceeding, I'm curious what @boruszak and @im2nguyen think.

Reply:
Here is more context about how I'm looking at the comment-based docs.

First, I do see the value. I have documented systems where the product ships with a commented up configuration file that the user duplicates and modifies to their implementation. It can be a useful way to present the information. 

For me, there are a few issues. As you pointed out, you can only provide the comments in HCL (and YAML), so we can't provide the examples for JSON configurations. We lose the rich text features, such as hyperlinks and text formatting, which might be useful in the explanations. Third, it's a deviation from our convention. Are users now expecting documentation within other example configuration files? 

I think the big thing, though, is that the information in this section is presented as a hybrid of conceptual and reference when we should present it as usage. There appears to be logical splits in the policy file where we could create "Define Vault managed PKI mount policies" section, for example. The section would include an explanation with links to the configuration reference items and an example. Or maybe it's a step-by-step process describing how to configure blocks of rules. When the information is presented as conceptual or reference, I think it reads as less actionable, so readers don't know if they should or have to do something.

I'll let Jeff and Tu chime in, though.

Reply:
If we went with the framing of "leave in comments where something may require thought/action", the only comment we'd want to leave in as the one about enabling root rotation, since best practices would suggest granting that permission for the minimum amount of time necessary when rotation is needed.